test/EndToEnd/vs.ps1

```diff
@@ -512,7 +512,9 @@ function New-FSharpLibrary {
         [string]$SolutionFolder
     )
 
-    New-Project FSharpLibrary $ProjectName $SolutionFolder
+    $project = New-Project FSharpLibrary $ProjectName $SolutionFolder
+    Wait-OnNetCoreRestoreCompletion $project
```

Comment:
Would it make sense for New-Project to do this wait when needed so tests don't need to remember to wait?

Reply:
Only SDK projects do automatic restore, and not all templates are SDK based PackageReference.

Reply:
Looking at the code for `Wait-OnNetCoreRestoreCompletion`, it waits for the `obj/project.assets.json` file to be created. However, this only happens for `PackageReference` projects. Non-SDK style projects that don't yet have any packages, or projects using packages.config, will never create this file. So, it won't work for all project types.